cub/thread/thread_operators.cuh

```diff
@@ -51,6 +53,33 @@ CUB_NAMESPACE_BEGIN
  * @{
  */
 
+/// @brief Inequality functor (wraps equality functor)
+template <typename EqualityOp>
+struct InequalityWrapper
+{
+  /// Wrapped equality operator
+  EqualityOp op;
+
+  /// Constructor
+  __host__ __device__ __forceinline__ InequalityWrapper(EqualityOp op)
+      : op(op)
+  {}
+
+  /// Boolean inequality operator, returns `t != u`
+  template <typename T, typename U>
+  __host__ __device__ __forceinline__ bool operator()(T &&t, U &&u)
+  {
+    return !op(std::forward<T>(t), std::forward<U>(u));
```

Comment:
I believe that should be 
```suggestion
    return !op(::cuda::std::forward<T>(t), ::cuda::std::forward<U>(u));
```

Reply:
This PR doesn't introduce any changes into mentioned code.

Reply:
I still believe we should use cuda::std functionality instead of plain `std::`

Technically `std::forward` is not annotated with `__device__` so...

Reply:
@miscco sure! The code originated when we missed libcu++ as a dependency. I meant that std::forward replacement can be done as a different PR, since it isn't related to "Use libcu++ function objects". I've checked and there seems to be not many places, so I can do it as part of this PR.